src/cr/cube/cube_slice.py

```diff
@@ -250,12 +265,43 @@ def is_double_mr(self):
         return self.dim_types == ['multiple_response'] * 2
 
     def scale_means(self, hs_dims=None, prune=False):
+        """Calculate scale means of the CubeSlice."""
         return self._cube.scale_means(hs_dims, prune)[self._index]
 
     @lazyproperty
     def ndim(self):
+        """Number of slice dimensions
+
+        Returns 2 if the origin cube has 3 or 2 dimensions.  Returns 1 if the
+        cube (and the slice) has 1 dimension.  Returns 0 if the cube doesn't
+        have any dimensions.
+        """
         return min(self._cube.ndim, 2)
 
     @lazyproperty
     def shape(self):
-        return self.as_array().shape
+        """Return the shape of the slice.
+
+        This property is anloguous to numpy's 'shape' property of ndarray. It
+        returns a tuple, returning the shape of the slice's 'as_array' method.
```

Comment:
It would always be preferable to avoid re-using the name of the method in its docstring. If a person doesn't know what "shape" means in this context, this docstring is going to make them work for it. So in this case, the second sentence might be better as: *"It returns a tuple of ints, each representing the length of a cube dimension, in the order those dimensions appear in the cube."*

I would add a blank line before the deprecated sentences. That topic is separate enough to warrant a distinct paragraph.